Rdmp.Dicom/CommandExecution/ExecuteCommandCFind.cs

```diff
@@ -0,0 +1,78 @@
+﻿using Rdmp.Core.Caching.Requests;
+using Rdmp.Core.Caching.Requests.FetchRequestProvider;
+using Rdmp.Core.CommandExecution;
+using Rdmp.Core.Curation;
+using Rdmp.Core.Curation.Data;
+using Rdmp.Core.Curation.Data.Cache;
+using Rdmp.Core.Curation.Data.DataLoad;
+using Rdmp.Core.DataFlowPipeline;
+using Rdmp.Core.Repositories;
+using Rdmp.Dicom.Cache.Pipeline;
+using ReusableLibraryCode.Progress;
+using System;
+using System.IO;
+
+namespace Rdmp.Dicom.CommandExecution
+{
+    class ExecuteCommandCFind : BasicCommandExecution, ICacheFetchRequestProvider
+    {
+        private BackfillCacheFetchRequest _request;
+        private CFindSource _source;
+
+        public ExecuteCommandCFind(IBasicActivateItems activator, string start, string end, string remoteAeUri, int remotePort, string remoteAeTitle, string localAeTitle, string outDir) : base(activator)
+        {
+            var startDate = DateTime.Parse(start);
+            var endDate = DateTime.Parse(end);
+
+            // Make something that kinda looks like a valid DLE load
+            var memory = new MemoryCatalogueRepository();
+            var lmd = new LoadMetadata(memory);
+
+            var dir = Directory.CreateDirectory(outDir);
+            var results = LoadDirectory.CreateDirectoryStructure(dir, "out", true);
+            lmd.LocationOfFlatFiles = results.RootPath.FullName;
+            lmd.SaveToDatabase();
+
+            var lp = new LoadProgress(memory, lmd);
+            var cp = new CacheProgress(memory, lp);
+
+            //Create the source component only and a valid request range to fetch
+            _source = new CFindSource
+            {
+                RemoteAEUri = new Uri("http://" + remoteAeUri),
+                RemoteAEPort = remotePort,
+                RemoteAETitle = remoteAeTitle,
+                LocalAETitle = localAeTitle,
+                TransferTimeOutInSeconds = 50000,
+                Modality = "ALL"
+            };
+            //<- rly? its not gonna pass without an http!?
```

Comment:
Yep. It's a URI, when what we really need is a hostname. That's in a few other places too, including UI I think - I'll squeeze in a refactoring of that later.